modules/private.php

```diff
@@ -0,0 +1,307 @@
+		}
+
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			return;
+		}
+
+		// Serve robots.txt for private blogs.
+		if ( is_object( $wp ) && ! empty( $wp->query_vars['robots'] ) ) {
+			return;
+		}
+
+		if ( is_user_logged_in() && self::is_private_blog_user( get_current_blog_id() ) ) {
+			return;
+		}
+
+		include JETPACK__PLUGIN_DIR . '/modules/private/private.php';
+
+		exit;
+	}
+
+	/**
+	 * Does not check whether the blog is private. Accepts blog and user in various types.
+	 * Returns true for super admins.
+	 *
+	 * @param int   $blog Current WordPress blod id..
+	 */
+	protected static function is_private_blog_user( $blog ) {
+		if ( is_numeric( $blog ) ) {
+			$blog_id = intval( $blog );
+		} elseif ( is_object( $blog ) ) {
+			$blog_id = $blog->blog_id;
+		} elseif ( is_string( $blog ) ) {
+			$fields  = array(
+				'domain' => $blog,
+				'path'   => '/',
+			);
+			$blog    = get_blog_details( $fields );
+			$blog_id = $blog->blog_id;
+		} else {
+			$blog_id = get_current_blog_id();
+		}
+
+		/**
+		 * Filter the capabilites a user needs to have to see the site
+		 *
+		 * @module private sites
+		 * @since 7.2
+		 *
+		 * @param string $cap The lowest capability a user needs to have
+		 */
+		$capability = apply_filters( 'jetpack_private_capability', 'read' );
+		return current_user_can_for_blog( $blog_id, $capability );
+	}
+
+	/**
+	 * Hides the blog's name on the login form for private blogs.
+	 */
+	static function privatize_blog_maybe_mask_blog_name() {
+		add_filter( 'bloginfo', array( __CLASS__, 'privatize_blog_mask_blog_name' ), 3, 2 );
+	}
+
+	/**
+	 * Replaces the the blog's "name" value with "Protected Blog"
+	 *
+	 * @see privatize_blog_maybe_mask_blog_name()
+	 * @param mixed $value The requested non-URL site information.
+	 * @param mixed $what  Type of information requested.
+	 */
+	static function privatize_blog_mask_blog_name( $value, $what ) {
+		if ( in_array( $what, array( 'name', 'title' ), true ) ) {
+			$value = __( 'Private Site', 'jetpack' );
+		}
+
+		return $value;
+	}
+
+	/**
+	 * Remove the privatize_blog_mask_blog_name filter
+	 */
+	static function remove_privatize_blog_mask_blog_name_filter() {
+		remove_filter( 'bloginfo', array( __CLASS__, 'privatize_blog_mask_blog_name' ) );
+	}
+
+	/**
+	 * Filters new comments so that users can't comment on private blogs
+	 *
+	 * @param array $comment Documented in wp-includes/comment.php.
+	 */
+	static function privatize_blog_comments( $comment ) {
+		self::privatize_blog( null );
+		return $comment;
+	}
+
+	/**
+	 * Extend the 'Site Visibility' privacy options to also include a private option
+	 **/
+	static function privatize_blog_priv_selector() {
+		?>
+		<style>
+			.jetpack-private__setting-disabled {
+				font-weight: bold;	
+				padding: 10px;
+			}
+			.option-site-visibility fieldset {
+				display: none;
+			}
+		</style>
+		</fieldset>
+		<div class="jetpack-private__setting-disabled highlight">
+			<?php
+				printf(
+					'%s <a href="%s">%s</a>',
```

Comment:
This is not ideal for translators, as not every language is going to be able to follow that sentence structure. It may be best to rely on `wp_kses` here to give a single sentence to translators. Same in other instances below.

Reply:
Reverted the change [suggested](https://github.com/Automattic/jetpack/pull/10333#discussion_r270569491) by @eliorivero in [d35a9cd](https://github.com/Automattic/jetpack/pull/10333/commits/d35a9cd9fc38b795c7ecd09055df47d1f7f922bc).